python/lsst/sconsUtils/state.py

```diff
@@ -192,6 +192,11 @@ def _initEnvironment():
         # We want to be explicit about the OS X version we're targeting
         #
         env['ENV']['MACOSX_DEPLOYMENT_TARGET'] = env['macosx_deployment_target']
+        # This flag is required for std::variant and std::filesystem
+        # on deployment platforms < 10.13 and 10.15
```

Comment:
So this is the right flag to set when building against conda libraries on macos.  The point is that we don't want to check against availability of features against the system libraries because we don't use them.  So the comment should be something like `This flag is required to use new features like std::variant and std::filesystem on older macos.  We build against conda libraries so we do not need to check if features are supported by system libraries.`

Reply:
So what we are really saying is that the deployment target is irrelevant when running inside conda.

Reply:
Almost.  (And our number has to be at least what conda-forge specifies which is 10.9).  But when building for osx-arm64 (not the subject of this ticket!) it does actually matter.